### PR TITLE
[Merged by Bors] - feat: dualising results in CategoryTheory.Functor.KanExtension.Adjunction

### PR DESCRIPTION
**Pull request by joelriou**
In this PR, we make a left/right synchronization of the API in the file `CategoryTheory.Functor.KanExtension.Adjunction`: missing definitions/lemmas for right Kan extensions are introduced.

---

- [x] depends on: #14256
- [x] depends on: #14258

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Functor/KanExtension/Adjunction.lean [hunk 1]
```diff
@@ -126,6 +139,115 @@ instance coreflective' [∀ (F : C ⥤ H), HasPointwiseLeftKanExtension L F] :
...
+noncomputable def ranCounit : L.ran ⋙ (whiskeringLeft C D H).obj L ⟶ (𝟭 (C ⥤ H)) where
+  app F := rightKanExtensionCounit L F
+  naturality {F₁ F₂} φ := by ext; simp [ran]
+
+instance (F : C ⥤ H) : (L.ran.obj F).IsRightKanExtension (L.ranCounit.app F) := by
+  dsimp [ran, ranCounit]
+  infer_instance
+
+/-- If there exists a pointwise right Kan extension of `F` along `L`,
+then `L.lan.obj G` is a pointwise right Kan extension of `F`. -/
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Functor/KanExtension/Adjunction.lean:166]:
```suggestion
/-- If there exists a pointwise right Kan extension of `F` along `L`,
then `L.ran.obj G` is a pointwise right Kan extension of `F`. -/
```

File: Mathlib/CategoryTheory/Functor/KanExtension/Adjunction.lean [hunk 2]
```diff
@@ -126,6 +139,115 @@ instance coreflective' [∀ (F : C ⥤ H), HasPointwiseLeftKanExtension L F] :
...
+noncomputable def isPointwiseRightKanExtensionRanCounit
+    (F : C ⥤ H) [HasPointwiseRightKanExtension L F] :
+    (RightExtension.mk _ (L.ranCounit.app F)).IsPointwiseRightKanExtension :=
+  isPointwiseRightKanExtensionOfIsRightKanExtension (F := F) _ (L.ranCounit.app F)
+
+variable (H) in
+/-- The right Kan extension functor `L.ran` is left adjoint to the
+precomposition by `L`. -/
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Functor/KanExtension/Adjunction.lean:174]:
```suggestion
/-- The right Kan extension functor `L.ran` is right adjoint to the
precomposition by `L`. -/
```

**Review by dagurtomas — APPROVED** [review]

**Comment by joelriou** [comment]
Thanks very much @dagurtomas for your review!

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**